Common/include/option_structure.hpp

```diff
@@ -174,34 +147,35 @@ enum AVERAGE_TYPE {
   AVERAGE_AREA = 1,     /*!< \brief Area-weighted average. */
   AVERAGE_MASSFLUX = 2  /*!< \brief Mass-flux weighted average. */
 };
-static const map<string, AVERAGE_TYPE> Average_Map = CCreateMap<string, AVERAGE_TYPE>
-("AREA", AVERAGE_AREA)
-("MASSFLUX", AVERAGE_MASSFLUX);
+static const MapType<string, AVERAGE_TYPE> Average_Map = {
+  MakePair("AREA", AVERAGE_AREA)
+  MakePair("MASSFLUX", AVERAGE_MASSFLUX)
+};
```

Comment:
This is what the map declaration semantics look like now, so that they can be switched on/off.
When map declarations are needed `#define ENABLE_MAPS` **before** including either `config/option_structure.hpp`, by default this is not default.
MakePair is a "do-nothing" macro when that flag is not defined, MapType is defined at the top as either std::map or a dummy empty class when the flag is not defined.